Android: Fix another regression with Secure.ANDROID_ID, and fix formatting and documentation of thirdparty code

akien-mga opened the pull request:
Issues introduced in #24145 + cleanup.

I'm not sure to what extent the actual changes in c54d986 are needed though, if we could do without them it would be much better. Or contribute the relevant changes upstream.

akien-mga committed:
Android: Fix another regression with Secure.ANDROID_ID

Regression from #24145, which was missed in #28146.

akien-mga committed:
Android: Resync Google licensing lib with upstream, unmodified

It had been synced with style changes (spaces -> tabs), not sure why
I accepted to merge it this way back then...

Synced with https://github.com/google/play-licensing/commit/eb57657f666363914085cdde49d875cf49f5ab06,
same as before.

Custom-changes will be reapplied in the next commit, if relevant.

akien-mga committed:
Android: Reapply changes to Google licensing lib from #24145

But document them better this time.

akien-mga committed:
Android: Resync Google expansion.downloader library with upstream, unmodified

Synced with https://github.com/google/play-apk-expansion/commit/9ecf54e5ce7c5a74a2eeedcec4d940ea52b16f0e.

akien-mga committed:
Android: Reapply custom changes to Google expansion.downloader lib

I don't know why they're needed, but readding for now to keep things
working as they were.

akien-mga committed:
Style: Don't apply clang-format to platform/android/java/src/com

akien-mga commented on platform/android/java/THIRDPARTY.md:
```diff
@@ -3,41 +3,16 @@
+Overwrite all files under:
 
-Modify those files to avoid compile error and lint warning:
+- `aidl/com/android/vending/licensing`
```
Regarding those, there was
```
// Android library projects do not yet support AIDL, so this has been
// precompiled into the src directory.
```
in their `.java` version included in the `src/` folder previously. But it seems to work fine now?

volzhs replied:
it works fine, because we are not compiling android **library** project.

akien-mga committed:
Android: Sync Google billing library with upstream, unmodified

Synced with https://github.com/googlesamples/android-play-billing/commit/7a94c6905a9c125518354c216b5c3094fde47ce1.

akien-mga commented:
Cherry-picked for 3.1.2. (Only the bugfix commit 071ebb1.)
